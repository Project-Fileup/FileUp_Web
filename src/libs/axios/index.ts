import axios from 'axios';
import createAxiosInstance from './createAxiosInstance';

export const commonAxios = createAxiosInstance({
  adapter: axios.defaults.adapter,
  headers: {
    'Cache-Control': 'no-cache',
  },
});

export const authAxios = createAxiosInstance();