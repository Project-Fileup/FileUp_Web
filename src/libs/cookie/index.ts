import { CookieSerializeOptions } from 'cookie';
import {
  getCookie,
  setCookie,
  deleteCookie,
  CookieValueTypes,
} from 'cookies-next';
import { ServerResponse, IncomingMessage } from 'http';

export type CookieOption = CookieSerializeOptions & {
  res?: ServerResponse;
  req?: IncomingMessage & {
    cookies?: Record<string, string> |
      Partial<Record<string, string>>;
  };
}

class Cookie {
  set<T>(key: string, data: T, options?: CookieOption): void {
    setCookie(key, data, options);
  }

  get(key: string): CookieValueTypes {
    return getCookie(key);
  }

  delete(key: string): void {
    deleteCookie(key);
  }
}

export default new Cookie();