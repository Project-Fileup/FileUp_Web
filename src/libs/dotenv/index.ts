const dotenv = {
  SERVER: process.env.NEXT_PUBLIC_SERVER as string,
  DEV_MODE: process.env.NODE_ENV === 'development',
} as const;

export default dotenv;