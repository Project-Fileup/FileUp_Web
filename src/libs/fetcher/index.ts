import axios from 'axios';
import { cacheAdapterEnhancer } from 'axios-extensions';
import createAxiosInstance from '@/libs/fetcher/createAxiosInstance';
import { requestHandler } from '@/libs/fetcher/request';
import { responseHandler } from '@/libs/fetcher/response';

const cacheAxios = createAxiosInstance({
  adapter: cacheAdapterEnhancer(axios.defaults.adapter!),
  headers: {
    'Cache-Control': 'no-cache',
  },
});

cacheAxios.interceptors.response.use(responseHandler);

const authAxios = createAxiosInstance();

authAxios.interceptors.request.use(requestHandler);
authAxios.interceptors.response.use(responseHandler);

export {
  cacheAxios,
  authAxios,
};