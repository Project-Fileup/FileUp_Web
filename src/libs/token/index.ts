import axios from 'axios';
import { Buffer } from 'buffer';
import dayjs from 'dayjs';
import { TokenKey } from '@/enums/auth';
import isServer from '@/utils/is-package/isServer';
import cookie, { CookieOption } from '../cookie';

export type JwtPayload = Record<string, unknown> & {
  iss?: string | undefined;
  sub?: string | undefined;
  aud?: string | string[] | undefined;
  exp?: number | undefined;
  nbf?: number | undefined;
  iat?: number | undefined;
  jti?: string | undefined;
}

class Token {
  set(tokenKey: string, token: string, options?: CookieOption): void {
    const expiredAt = dayjs().add(7, 'days').toDate();

    cookie.set(tokenKey, token, {
      ...options,
      expires: expiredAt,
    });
  }

  get(tokenKey: string): string {
    switch (tokenKey) {
      case TokenKey.ACCESS_TOKEN:
        if (isServer()) {
          const axiosToken = axios.defaults.headers.common[TokenKey.REQUEST_TOKEN];

          return axiosToken?.replace('Bearer ', '');
        }
    }

    return cookie.get(tokenKey) as string;
  }

  decode(token: string): JwtPayload | undefined {
    try {
      const base64Payload = token?.split('.')[1];
      const payload = Buffer.from(base64Payload, 'base64');

      return JSON.parse(String(payload)) as JwtPayload;
    } catch (error) {

    }
  }

  clear(): void {
    cookie.delete(TokenKey.ACCESS_TOKEN);
    cookie.delete(TokenKey.REFRESH_TOKEN);

    axios.defaults.headers.common[TokenKey.REQUEST_TOKEN] = '';
  }
}

export default new Token();