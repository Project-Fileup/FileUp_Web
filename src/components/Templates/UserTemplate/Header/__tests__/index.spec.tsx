import { render } from '@/libs/testing';
import Header from '..';

describe('헤더 컴포넌트 테스트', () => {
  it('스냅샷 출력 결과물과 동일하게 렌더링된다.', () => {
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});