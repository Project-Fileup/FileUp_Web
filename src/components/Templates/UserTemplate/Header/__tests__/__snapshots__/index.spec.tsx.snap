// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`헤더 컴포넌트 테스트 스냅샷 출력 결과물과 동일하게 렌더링된다. 1`] = `
<div>
  <div />
</div>
`;
